core/src/main/scala/chisel3/BitsImpl.scala

```diff
@@ -528,7 +528,7 @@ private[chisel3] trait ResetTypeImpl extends Element { self: Reset =>
 
   def cloneType: this.type = Reset().asInstanceOf[this.type]
 
-  override def litOption = None
+  override def litOption: None.type = None
```

Comment:
This is similar to the discussion here: https://github.com/chipsalliance/chisel/pull/4403#discussion_r1763647813

We actually want `-Xsource-features:infer-override` because we would rather the type be `Option[BigInt]` than `None.type`. **However**, unlike in the discussion on the previous PR, this change matters so needs to be documented in the release notes.

Looking through, there aren't very many where it matters so that's good, but for any public API, we need to document that the type has changed, e.g.

* `chisel3.ResetType.litOption` type changed from `None.type` to `Option[BigInt]`

Reply:
As another example, see the changes I made to the Release Notes in https://github.com/chipsalliance/chisel/pull/4430.
The APIs in svsim matter less since they are commonly used, but many of the APIs here in Chisel matter, like this one.